README.md

```diff
@@ -101,44 +101,116 @@ This directory contains a `Makefile` and a few directories.
 
 ### Templating
 We are using the [Jinja templating language](http://jinja.pocoo.org/docs/latest/templates/).
-As a convenience we have added some custom syntax to make it more convenient to include auditor notes.
+We have also added a few [jinja extensions](http://jinja.pocoo.org/docs/2.10/extensions/) 
+to simplify document management.
+These extensions are optional.
+They are included by listing them in `system.yml`.
+For example
+```yaml
+# List optional extensions
+extension_load_list:
+  - 'rdm.md_extensions.audit_notes.AuditNoteExtension'
+  - 'rdm.md_extensions.section_numbers.SectionNumberExtension'
+  - 'rdm.md_extensions.vocabulary_extension.VocabularyExtension'
+```
+That list is also a convenient place to load any other jinja extensions.
+
+In addition to the optional extensions an object `first_pass_output` is added as a jinja variable.
+It has two useful properties:
+  * `first_pass_output.source` has the complete output of an initial trial generation of the document.
+  * `first_passoutput.lines` has that same output as a more convenient list of lines.
+
+### Auditor Notes Extension
+As a convenience we have added some features to make it more convenient to include auditor notes.
 Auditor notes are references that will be convenient to an auditor 
 but add a lot of extraneous information to others.
 In the `system.yml` file `auditor_notes` can be set true or false.
```

Comment:
To summarize the changes we discussed earlier:
- We replace the `system.auditor_notes` boolean with the current value of `auditor_note_formats`, and remove the `auditor_note_formats`  value completely.
- We update the templates to remove any multi-line auditor notes.
- We remove the need for `if` statements in the templates, surrounding the `{% auditor_notes system.auditor_notes %}` tag.

Once we make these changes, we will need to update the documentation.

To be honest, I would de-prioritize documentation for now, since the libraries' API is likely to change frequently until we get to the 1.0 release.  I think it will cost a lot of time updating the documentation as we alter the API, and I don't want the existence of documentation to discourage us from playing and/or simplifying the API.  I think the documentation you have written is great, and as soon as we have settled on an API, we can write similar docs.

I say this, partly from our experience with the slicer library.  We have continued improving the library with each new project.  We updated the documentation for a while during emass, but eventually, we gave up because it was too much work with no real value added at the moment.